### ISSUE TITLE
Support for Backup infrastructure

### ISSUE THREAD

**Issue by GeorgeBellTMH** [aws-amplify/amplify-cli#5972]
All services that store data should have a way to setup backups

amplify add storage
- do you want to create automatic backups?

amplify add api
- dynamo
- do you want to create automatic backups?

amplify add auth
- do you want to create automatic backups?

**Comment by GeorgeBellTMH** [comment]
For API it may be useful to have @backup(frequency: xxx, retainDuration:) that can be defined on each table. Or to be able to use point in time for any or all tables. 

For auth it would be useful to be able to backup the user information and group membership (even if passwords aren't backed up) with the option of selecting which custom fields (or all of them) get backed up. Optionally this should be able to be backed up in an encrypted format. 

For storage it would be useful to have rules around how the backups are made, concluding wildcards to define what to backup, how long to store, what type of s3 storage to use etc. 

All backups would ideally end up in an S3 bucket on either the same or a different region. Also it would be nice if they could be downloaded and stored outside of Amazon if necessary.

**Comment by giorgiocarniel** [comment]
Any news?

**Comment by danielesalvatore** [comment]
Up

**Comment by swatipraskar** [comment]
Yes it’s important. Any news..For any serious App, you cannot go live without backup and DR features..
Ideally, 
amplify dr
How much RTO ( recovery time objective )
How much RPO ( recovery point objective )

Finally need dr console feature to switch Prod domain to DR

Is there anything in the pipeline.. If not, what is the work around until final solution is in place?

**Comment by swatipraskar** [comment]
Yes it’s important. Any news..For any serious App, you cannot go live without backup and DR features..
Ideally,
amplify dr
How much RTO ( recovery time objective )
How much RPO ( recovery point objective )

Finally need dr console feature to switch Prod domain to DR

Is there anything in the pipeline.. If not, what is the work around until final solution is in place?

**Comment by swatipraskar** [comment]
Up too

**Comment by swatipraskar** [comment]
Up

**Comment by droidgeek1** [comment]
Up

**Comment by droidgeek1** [comment]
Also add Disaster Recovery mechanisms

**Comment by CodySwannGT** [comment]
As someone who was just bitten by this, it would really be nice to have backups be part of amplify.

1) DynamoDB turns on Point in Time Recovery
2) S3 replicates to another bucket
3) Cognito replicates to another pool

The backups ARE NOT deleted if and when the amplify app is deleted.

And add a command like 

$ amplify restore-from-backup

**Comment by cfbo** [comment]
This feature would be really essential for us. We are building a solution that will soon need to be released in production soon and it's completely built on Amplify. 
I have to admit that I have only started on AWS/Amplify over the last couple of months, but I don't see how one can build a production-ready application without this feature and some goes for the topic reported in #7672.
While waiting for this to be hopefully implemented, if anyone can share the solution they are using it would be super helpful. I was going to use AWS Backup, configuring a backup plan based on tags (for graphql API/DynamoDB).
Just a thought: compared to implementing #7672 which I can understand would be quite hard, taking care of the feature request looks like an easy win that would provide so much value to the community.

**Comment by GeorgeBellTMH** [comment]
1. I wonder how much critical data is stored by amplify and has ineffective or non-existent data backups
2. I wonder how many people have abandoned amplify because it doesn't support this out of the box
3. I wonder how much money amazon has missed out on because of 1 and 2...

**Comment by GeorgeBellTMH** [comment]
If the amplify team is reading, an RFC on Data Backup/Site Backup/Failover would be great!

1. Ability to setup/configure backups for Dynamo, Cognito, Aurora, Pinpoint, S3 - within a datacenter, or to a second data centre from CLI
2. Ability to restore above backups from CLI
3. Ability to setup multi data-centre active/active or active/passive configurations from within CLI
4. Optional encryption of backups
5. Retention periods etc.

**Comment by cfbo** [comment]
It would be great to get some feedback on this from the amplify team, on whether they have any plan to implement this feature in the future or whether it's never going to happen.

**Comment by cfbo** [comment]
Any update?

**Comment by julien-tamade** [comment]
Up too

**Comment by octaneC8H18** [comment]
Any update?

**Comment by danielesalvatore** [comment]
Waiting for the feature to be available in AWS Amplify, I paste here my solution implemented using AWS CDK

`
        const plan = backup.BackupPlan.dailyWeeklyMonthly5YearRetention(
        this,
        'Plan'
      )
       const branchName = process.env.AWS_BRANCH
       const dynamodbTableHash = 'your-hash' // dynamodb table names are like 'name-hash-branch'

      let tables = []
      let tableNames = [
        'table1',
        'table2',
         // list here your tables names. I did not find a way to dynamically read it as a Amplify output
      ]

      tables = tableNames.map((name) => {
        const dynamodbTableId = `${name}table`
        const dynamodbTableName = `${name}${dynamodbTableHash}${branchName}`
        const dynamodbTable = dynamodb.Table.fromTableName(
          this,
          dynamodbTableId,
          dynamodbTableName
        )
        return backup.BackupResource.fromDynamoDbTable(dynamodbTable)
      })

      let resources = [
        backup.BackupResource.fromTag('stage', 'prod'), // All resources that are tagged stage=prod in the region/account
      ].concat(tables) // DynamoDB tables

      plan.addSelection('BackupSelection', {
        resources,
      })
`
For completeness, I dynamically populate `dynamodbTableHash` using a switch-case based on `branchName` to target the different Amplify environments.
